packages/react-scripts/config/webpackDevServer.config.js

```diff
@@ -68,7 +68,7 @@ module.exports = function(proxy, allowedHost) {
     hot: true,
     // It is important to tell WebpackDevServer to use the same "root" path
     // as we specified in the config. In development, we always serve from /.
-    publicPath: '/',
+    publicPath: paths.servedPath.slice(0, -1),
```

Comment:
I have been messing around with this locally and I have found that 
further down in this file needs to be updated as well, something similar to:
```javascript
    historyApiFallback: {
      // Paths with dots should still use the history fallback.
      // See https://github.com/facebook/create-react-app/issues/387.
      disableDotRule: true,
      index: paths.servedPath,
    },
```
otherwise when you go to pages other than the home page with react router setup and refresh it will fail to load files.